cypress/integration/core_opensearch_dashboards/opensearch_dashboards/apps/query_enhancements/export_as_csv.spec.js

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import Papa from 'papaparse';
+import {
+  INDEX_WITH_TIME_1,
+  PATHS,
+  DATASOURCE_NAME,
+  INDEX_WITHOUT_TIME_1,
+  DatasetTypes,
+  QueryLanguages,
+} from '../../../../../utils/constants';
+import {
+  getRandomizedWorkspaceName,
+  setDatePickerDatesAndSearchIfRelevant,
+} from '../../../../../utils/apps/query_enhancements/shared';
+import { prepareTestSuite } from '../../../../../utils/helpers';
+import {
+  generateExportAsCsvIndexPatternTestConfigurations,
+  getFirstRowForDownloadWithFields,
+  getFirstRowTimeForSourceDownload,
+  getHeadersForDownloadWithFields,
+  getHeadersForSourceDownload,
+  getQueriedCountForLanguage,
+} from '../../../../../utils/apps/query_enhancements/export_as_csv';
+import moment from 'moment';
+import path from 'path';
+
+const workspaceName = getRandomizedWorkspaceName();
+
+const runSavedSearchTests = () => {
+  describe('Export as CSV', () => {
+    beforeEach(() => {
+      // Load test data
+      cy.osd.setupTestData(
+        PATHS.SECONDARY_ENGINE,
+        [
+          `cypress/fixtures/query_enhancements/data_logs_1/${INDEX_WITH_TIME_1}.mapping.json`,
+          `cypress/fixtures/query_enhancements/data_logs_1/${INDEX_WITHOUT_TIME_1}.mapping.json`,
+        ],
+        [
+          `cypress/fixtures/query_enhancements/data_logs_1/${INDEX_WITH_TIME_1}.data.ndjson`,
+          `cypress/fixtures/query_enhancements/data_logs_1/${INDEX_WITHOUT_TIME_1}.data.ndjson`,
+        ]
+      );
+
+      // Add data source
+      cy.osd.addDataSource({
+        name: DATASOURCE_NAME,
+        url: PATHS.SECONDARY_ENGINE,
+      });
+
+      // Create workspace
+      cy.deleteAllWorkspaces();
+      cy.visit('/app/home');
+      cy.osd.createInitialWorkspaceWithDataSource(DATASOURCE_NAME, workspaceName);
+      cy.createWorkspaceIndexPatterns({
+        workspaceName: workspaceName,
+        indexPattern: INDEX_WITH_TIME_1,
+        timefieldName: 'timestamp',
+        dataSource: DATASOURCE_NAME,
+        isEnhancement: true,
+      });
+      cy.createWorkspaceIndexPatterns({
+        workspaceName: workspaceName,
+        indexPattern: INDEX_WITHOUT_TIME_1,
+        timefieldName: '',
+        dataSource: DATASOURCE_NAME,
+        isEnhancement: true,
+        indexPatternHasTimefield: false,
+      });
+    });
+
+    afterEach(() => {
+      cy.deleteWorkspaceByName(workspaceName);
+      cy.osd.deleteDataSourceByName(DATASOURCE_NAME);
+      cy.osd.deleteIndex(INDEX_WITH_TIME_1);
+      cy.osd.deleteIndex(INDEX_WITHOUT_TIME_1);
+    });
+
+    generateExportAsCsvIndexPatternTestConfigurations().forEach((config) => {
+      it(`should be able to export for ${config.saveName}`, () => {
+        cy.navigateToWorkSpaceSpecificPage({
```

Comment:
I think for the test can we add two more scenarios 1) add a filter or use a query to filter then the number shown in the download can be updated? for example after adding filter we should see the number to download also updated to 127 2) In the advanced settings, if you reset the default size, then load discover, the number should be updated to the settings
<img width="1506" alt="Screenshot 2025-02-06 at 6 21 00 PM" src="https://github.com/user-attachments/assets/84384453-6a2d-43f3-bddd-b389bd61212c" />

<img width="1704" alt="Screenshot 2025-02-06 at 6 16 48 PM" src="https://github.com/user-attachments/assets/665085de-1e84-464c-9324-8755e4e20d77" />

Reply:
sure I can add that test, that's a good idea